### PR TITLE
Setup association

### PR DESCRIPTION
**Pull request by Estherstarr**
This PR contains the creation of the Classroom, Book & Rental classes for our School Library and created the Associations between them.
A school library app to record what books are in the library and who borrows them. The app that you will create will allow you to: add new students or teachers, add new books and save records of who borrowed a given book and when.

Highlights:

- [ ] Created a class Classroom with the following:
- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).
- [ ] Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- [ ] Created the has-many side (a classroom has many students).
- [ ] Created the belongs-to side (a student belongs to a classroom).
- Made sure that when adding a student to a classroom it also sets the classroom for the student.
- Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- [ ] Created a class Book with the following:
- @title and @author instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor).
- [ ] Created a class Rental with the following:
- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).
- [ ] Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- [ ] Created the has-many side of Book and Rental (a book has many rentals).
- [ ] Created the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Created the has-many side of Person and Rental (a person has many rentals).
- [ ] Created the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Defines a person class
```
Comment (OLIPLICHE) [inline person.rb:3]: - Please make sure that your Person class inherits from Nameable: It should be something like this `class Person < Nameable`
- Also, make sure that this class has a method `correct_name implemented`:
- Please these are the project requirements that you need to follow carefully. [Open here to learn more about Decorator pattern in Ruby ](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Estherstarr, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Estherstarr) [reply person.rb:3]: In this PR
- [ ] Person class inherits from Nameable
- [ ] Person class has a method correct_name implemented:

**Review by sumairq — APPROVED** [review]
### Hi @Estherstarr  , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Estherstarr** [comment]
This pull request contains both the decorator and setup associations

**Pull request merged**